### PR TITLE
update to go1.23 in `main`

### PR DESCRIPTION
**Pull request by jbardin**
Release notes: https://go.dev/doc/go1.23

Code changes were because some updates to staticcheck were catching more errors. Most of these were format functions without a format string, but rather than adding more exceptions I just fixed them all. This did also catch some incorrectly wrapped errors however.

### CODE REVIEW

**Comment by jbardin** [comment]
Among other minor updates, we also have #35426 pending use of the new range over functions feature.

**Comment by jbardin** [comment]
I'm also doing concurrent work in the nested backend modules which will conflict here, but will rebase before merging.

**Review by liamcervante — APPROVED** [review]
Change looks fine to me, leave it up to you about the risks wrt to the issues the AWS provider is seeing.

**Pull request merged**
